az-cvm-vtpm/src/vtpm/mod.rs

```diff
@@ -129,7 +129,23 @@ pub struct Quote {
     pcrs: Vec<Vec<u8>>,
 }
 
+fn pad<const T: usize>(input: &[u8]) -> [u8; T] {
+    let mut output = [0; T];
+    let len = input.len();
+    if len > T {
+        output.copy_from_slice(&input[..T]);
+    } else {
+        output[..len].copy_from_slice(input);
+    }
+    output
+}
+
 impl Quote {
+    /// Retrieve sha256 PCR values from a Quote
+    pub fn pcrs_sha256(&self) -> Vec<[u8; 32]> {
+        self.pcrs.iter().map(|x| pad(x)).collect()
```

Comment:
why do you need to pad? all entries should be sha256 length so we should only need to check length and copy into the fixed size array.

The pad function also does both truncation or padding with zeros, neither of which seem correct.

Reply:
As in - we might hide errors here.

Reply:
I think the idea was that this fn shouldn't return a Result, because the quote should already be correct, and this would be weird. so any sort of check would have to result in a panic.

however, this was based on the assumption that the library just returns unsized vectors, but turns out they do have some convenience trait impls to help w/ (fallibly) conversion in sized arrays, so we should change the internal representation to [u8; 32].